compass/landice/tests/antarctic/mesh.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+import netCDF4
+import xarray
+from matplotlib import pyplot as plt
+
+from mpas_tools.mesh.creation import build_planar_mesh
+from mpas_tools.mesh.conversion import convert, cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+from mpas_tools.io import write_netcdf
+from mpas_tools.logging import check_call
+
+from compass.step import Step
+from compass.model import make_graph_file
+from compass.landice.mesh import gridded_flood_fill, \
+                                 set_rectangular_geom_points_and_edges, \
+                                 set_cell_width, get_dist_to_edge_and_GL
+
+
+class Mesh(Step):
+    """
+    A step for creating a mesh and initial condition for antarctica test cases
+
+    Attributes
+    ----------
+    mesh_type : str
+        The resolution or mesh type of the test case
+    """
+    def __init__(self, test_case):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        mesh_type : str
+            The resolution or mesh type of the test case
+        """
+        super().__init__(test_case=test_case, name='mesh')
+
+        self.add_output_file(filename='graph.info')
+        self.add_output_file(filename='Antarctic_1to10km.nc')
```

Comment:
It's a little tricky to do but maybe we want the name of the output mesh to include the actual requested minimum and maximum resolution, rather than these hard-coded values?  Something to discuss.

Reply:
I think something like this would work, but is ugly to look at:
`      config = self.config
        section = config['humboldt']
        min_spac = float(section.get('min_spac'))
        max_spac = float(section.get('max_spac'))
        min_spac_str = f"{round(min_spac/1000, 2):.1f}"
        max_spac_str = f"{round(max_spac/1000, 2):.1f}"
        if min_spac_str[-1] == '0':
             min_spac_str_trimmed =  min_spac_str[0:-2]
        else:
             min_spac_str_trimmed = min_spac_str

        if max_spac_str[-1] == '0':
             max_spac_str_trimmed =  max_spac_str[0:-2]
        else:
             max_spac_str_trimmed = max_spac_str

        output_file_name = 'Antarctic_{}to{}km.nc'.format(min_spac_str_trimmed, max_spac_str_trimmed))` 
                       
We could add a function to `compass.landice.mesh` that does this kind of filename formatting.

Reply:
Yes, that's roughly what I was thinking of.  But a serious complication is that the file name isn't known until the `run()` method because a user could alter the config options before then.  That makes it pretty hard to define the output name either in the `__init__()` method or the `setup()` method.  Maybe in the end it would be better not to have the resolution in the file name at all?  Or maybe we're fine with hard coding the resolution even if a user might change it.  I just want to have the discussion so this doesn't happen without thinking it through.

Reply:
Okay, I see what you mean. Our convention for MALI mesh names includes the resolution, but also other things like the date and a version number because of all the iterations we end up doing with the optimization and initial condition datasets, so I'd be totally fine with not including resolution in the output file name within COMPASS. @matthewhoffman, what do you think?

Reply:
I agree we could leave it out given the versioning system we have been (mostly) using successfully.  But this is a valid concern that we may want to revisit down the load if/when our mesh generation workflow is truly end to end.